File: MojaCoder/pdc001/f.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 0; i < (n); ++i)
#define rep1(i, n) for(int i = 1; i < (n); ++i)
using namespace std;
using ll = long long;

bool is_prime(int n){
    if(n == 1) return false;
    
    for(int i = 2; i*i <= n; ++i){
        if(n%i == 0) return false;
    }

    return true;
}

int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    vector<ll> primes;
    rep1(i, 10000000){
        if(is_prime(i)) primes.push_back(i);
    }

    ll a, b;
    cin >> a >> b;

    ll res = 0;
    for(ll prime: primes){
        ll base = 1;
        while(1){
            ll num = pow(prime, base);
            if(a <= num && num <= b){
                res = max(res, base);
            }
            ++base;
            if(num > b) break;
        }
    }

    cout << res << endl;
    

    return 0;
}

File: MojaCoder/pdc001/f.py
def isprime(n):
    if n == 1: return False
    for i in range(2, n):
        if i*i > n: break
        if n%i == 0: return False
    return True

primes = []

for i in range(1, 1000000):
    if isprime(i): primes.append(i)

print(len(primes))
# print(primes[: 10])
a, b = map(int, input().split())

res = 0
for prime in primes:
    base = 1

    while True:
        num = prime ** base
        if a <= num <= b:
            res = max(res, base)
        base += 1
        if num > b:
            break

print(res)